lib/mongo/connection/node.rb

```diff
@@ -73,6 +73,16 @@ def connect
       end
     end
 
+    def socket
+      if @socket && @socket.pid != Process.pid
```

Comment:
If the process is not forked, `@socket.pid` will *always* return `nil`, which will cause the socket to be closed and reopened on *every* request, thus causing a significant negative impact on performace.